docs/tutorial_docker.md

```diff
@@ -1,10 +1,100 @@
+# Running FragPipe using Docker and Singularity
+
+This tutorial covers how to use **Docker** and **Singularity** to run FragPipe. We will run the `fcyucn/fragpipe:<version>` container, first with Docker, then converting it to a Singularity image.
+
+## Prerequisites for Docker and Apptainer (Linux)
+
+Before setting up Docker and Apptainer on Linux, ensure the following:
+
+- **Operating System**: A Linux distribution (e.g., Ubuntu, CentOS, Debian).
+- **Docker Installation**: Install Docker by following the [official guide for Linux](https://docs.docker.com/engine/install/).
+- **Apptainer Installation**: Install Apptainer from the [Apptainer documentation](https://apptainer.org/docs/admin/main/installation.html).
+- **Using Singularity Commands**: Apptainer supports Singularity commands. After installation, you can use `singularity` as the command-line interface (`apptainer` binary serves as `singularity`).
+
+- **User Privileges**: Ensure you have `sudo` privileges for installation.
+
+## Running FragPipe with Docker
+
+1. **Pull and run the Docker container:**
+
+   To run the FragPipe tool directly using Docker, execute:
+
+   ```bash
+   docker run fcyucn/fragpipe:<version> fragPipe-<version>/fragpipe/bin/fragpipe --help
+   ```
+
+   This command will:
+   - Pull the `fcyucn/fragpipe:<version>` image from Docker Hub.
+   - Run the `fragpipe` command inside the container, displaying the help options.
+
+2. **Mounting the Host Directory to the Docker Container:**
+
+   One of the most useful features of Docker is the ability to mount a host directory inside the container. This is especially helpful when you need to access data files or output results from FragPipe.
+
+   You can mount a local directory (e.g., `/path/to/your/data`) to the Docker container using the `-v` option:
+
+   ```bash
+   docker run -v /path/to/your/data:/mnt/data fcyucn/fragpipe:<version> fragPipe-<version>/fragpipe/bin/fragpipe --help
+   ```
+
+   - `/path/to/your/data`: The path on your host machine where your data is located.
+   - `/mnt/data`: The directory inside the container where the host directory will be mounted.
+   
+   Now, any files placed in `/path/to/your/data` will be accessible from `/mnt/data` inside the Docker container. This allows you to read input files and save output directly from/to your local machine while running FragPipe in Docker.
+
+3. **Running FragPipe Inside the Docker Container:**
+
+   To run FragPipe inside the container, mount the required directories (for example, where your data or project files are located), then execute the FragPipe tool. You can use the following command:
+
+   ```bash
+   docker run -v /path/to/your/data:/mnt/data -it fcyucn/fragpipe:<version> /fragpipe_bin/FragPipe-<version>/fragpipe/bin/fragpipe -config /mnt/data/config_file.fpconfig

```

Comment:
What is the `-config` flag for? And what is the format of the fpconfig file? Why not just provide FragPipe's arguments?